spacepy/pybats/bats.py

```diff
@@ -1226,65 +1343,66 @@ def gradP_regular(self, cellsize=None, dim1range=-1, dim2range=-1):
         if self.gridtype != 'Regular':
             if not cellsize:
                 raise ValueError('Grid must be regular or ' +
-                                  'cellsize must be given.')
+                                 'cellsize must be given.')
             self.regrid(cellsize, dim1range=dim1range, dim2range=dim2range)
 
         # Order our dimensions alphabetically.
         newvars = []
         for key in sorted(self.grid.keys()):
             newvars.append('gradp'+key)
 
-        dx=self.resolution*6378000.0 # RE to meters
-        p =self['p']*10E-9        # nPa to Pa
-        self[newvars[0]], self[newvars[1]]=gradient(p, dx, dx)
-        self['gradp']=sqrt(self[newvars[0]]**2.0 + self[newvars[1]]**2.0)
+        dx = self.resolution * 6378000.0  # RE to meters
```

Comment:
`RE` is defined at module level as 6371000 m, which conflicts with the conversion here. Why is 6378000 used here? Or conversely, why is the module-level `RE` using a different value?

6378000 is close to, but not, the WGS84 equatorial radius (that'd be 6378137).
6371000 is the volumetric mean radius, IIRC, which is used in codes like IGRF so perhaps it's appropriate here?